update_gh_pages.sh

```diff
@@ -1,10 +1,37 @@
-#!/bin/bash
+#!/bin/sh
+
+# Make sure we're not echoing any sensitive data
+set +x
+
 set -e
 
+opam install -y caml2html
 make html
-rm -fr /tmp/doc
-rsync -av doc/gen/ /tmp/doc/
-git checkout gh-pages
-mv .git /tmp/doc/
-rsync --delete -av /tmp/doc/ .
 
+if [ -z "$TRAVIS" -o "$TRAVIS_PULL_REQUEST" != "false" -o "$TRAVIS_BRANCH" != "master" ]; then
+  echo "This is not a push Travis-ci build on the master branch, doing nothing..."
```

Comment:
Typo: s/build/built/. I would rephrase it: "This pull request does not target the master branch - ignoring it"

Reply:
I think this is accurate, as this must not run for a pull request at all; only for builds running in Travis for commits pushed to the master branch.

Reply:
I've removed the word `push`, now it sounds better.